File: backend/app/demo_users.py
"""Preset demo users for the public MemGuard demo.

These exist so judges/visitors can reset the whole demo cleanly by switching
users, without wiping real data or needing any login.
"""

DEMO_USERS: list[dict[str, str]] = [
    {"id": "demo_alice", "label": "Alice (Pro plan, IST)"},
    {"id": "demo_bob", "label": "Bob (fresh, no memories yet)"},
    {"id": "demo_carol", "label": "Carol (poisoning scenario seed)"},
]

File: frontend/lib/demoUsers.ts
import type { DemoUser } from "./types";

/**
 * Fallback preset users, used only if the backend /demo/users call fails
 * (e.g. offline local dev without the backend running yet). Keeps the
 * dropdown usable so Screen A never renders empty.
 */
export const FALLBACK_DEMO_USERS: DemoUser[] = [
  { id: "demo_alice", label: "Alice (Pro plan, IST)" },
  { id: "demo_bob", label: "Bob (fresh, no memories yet)" },
  { id: "demo_carol", label: "Carol (poisoning scenario seed)" },
];
